contracts/FlowIDTableStaking.cdc

```diff
@@ -56,12 +36,18 @@ pub contract FlowIDTableStaking {
     pub event NewWeeklyPayout(newPayout: UFix64)
     pub event NewStakingMinimums(newMinimums: {UInt8: UFix64})
 
+    /// Indicates if the staking auction is currently active
+    pub var stakingEnabled: Bool
+
     /// Holds the identity table for all the nodes in the network.
     /// Includes nodes that aren't actively participating
     /// key = node ID
-    /// value = the record of that node's info, tokens, and delegators
     access(contract) var nodes: @{String: NodeRecord}
 
+    access(contract) var stakingKeyClaimed: {String: Bool}
+    access(contract) var networkingKeyClaimed: {String: Bool}
+    access(contract) var networkingAddressClaimed: {String: Bool}
```

Comment:
I'd recommend not storing these dictionaries in the contract itself, as all this data will need to be decoded each time a transaction/script imports and uses the `FlowIDTableStaking` contract value.

For now it's better to store this data in the contract owner's account storage in separate storage paths, and maybe add some helper functions to access the data easily.

Reply:
We learned this the hard way in the Top Shot contract, which stores all plays and sets in the contract, so each time the Top Shot contract is used, all the data needs to be unnecessarily loaded from storage and decoded, which is expensive